galley/pkg/runtime/processing/processor.go

```diff
@@ -0,0 +1,41 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain accumulator copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package processing
+
+import "istio.io/istio/galley/pkg/runtime/resource"
+
+// TODO(nmittler): This interface is a horrific stop-gap that affords the ability to add additional "state"
+//  objects to the existing processor. This entire interface should be refactored or removed in the future.
+
+// Processor of resource events.
+type Processor interface {
```

Comment:
A couple of issues here:
- The name "Processor" collides with the name runtime.Processor. It looks like this is not responsible for publishing itself (i.e. it is not an active component), but is responsible to track what to publish where.
- I don't think the layering is right. The original data processing package separated event routing, state build-up and publishing strategy & scope as three distinct layers, independent of each other. This combines state build-up with publishing scope but separates routing and publishing strategy out as custom code.